torch/autograd/profiler.py

```diff
@@ -283,6 +293,7 @@ def __enter__(self):
         if self.entered:
             raise RuntimeError("Profiler context manager is not reentrant")
         _enable_dynamo_cache_lookup_profiler(True)
+        _set_is_profiler_enabled(True)
```

Comment:
I'm not very familiar with all the entry points to the profiler. Is this really the only place where the profiler can ever be enabled?

Reply:
It's probably better if we put it at all the callsites of `torch._C._autograd.enable_profiler` - I'll change it. I think the difference is that `with emit_nvtx():` and `with emit_itt():` would not be included in the current implementation